include/extractor/edge_based_graph_factory.hpp

```diff
@@ -182,7 +188,15 @@ class EdgeBasedGraphFactory
                                    const std::string &turn_penalties_index_filename,
                                    const bool generate_edge_lookup);
 
-    void InsertEdgeBasedNode(const NodeID u, const NodeID v);
+    // Mapping betweenn the node based graph u,v nodes and the edge based graph head,tail edge ids.
+    // Required in the osrm-partition tool to translate from a nbg partition to a ebg partition.
+    struct Mapping
+    {
+        NodeID u, v;
+        EdgeID head, tail;
```

Comment:
`head` and `tail` can be a little bit confusing here because they typically reference the start/target for an `arc` (edge based edge). However there is no arc from `tail` to `head` and both can be heads and tails for different arcs (there are typically arcs both from and to `head` and `tail`).

`{forward,backward}_edge_based_node` might be more precise but maybe you have another idea for a name.